libs/SmartStore/SmartStore/Classes/SFSmartStore.m

```diff
@@ -1351,6 +1341,19 @@ - (void)registerSoupWithSpec:(SFSoupSpec*)soupSpec withIndexSpecs:(NSArray*)inde
     }
     [self insertIntoSoupIndexMap:soupIndexMapInserts withDb:db];
     
+    // LogEvent
+    NSMutableArray<NSString *> *features = [[NSMutableArray alloc] init];
+    if (soupUsesJSON1) {
+        [features addObject:@"JSON1"];
+    }
+    if (soupUsesExternalStorage) {
+        [features addObject:@"ExternalStorage"];
+    }
+    if ([self hasFts:soupSpec.soupName withDb:db]) {
```

Comment:
You are right, we should log the event if the registration successfully goes through not before.
For soupUsesFts, calling the static hasFts method on SFSoupIndex is the easiest/cheapest way.